### PR TITLE
Refactor Parameters for Joining Calls

### PR DESCRIPTION
**Pull request by iaulakh**
## Purpose

* Refactor Parameters for Joining Calls to prepare for future Teams Interop changes.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/JoinCallConfig.java [hunk 1]
```diff
@@ -16,14 +16,14 @@ public JoinCallConfig(
             final boolean isMicrophoneMuted,
             final boolean isCameraOn,
             final String displayName) {
-        this.groupId = groupId;
+        this.joinId = groupId;
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/JoinCallConfig.java:19]: Do we still have a `groupId` somewhere?

**Review by waynemo — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/JoinCallConfig.java (hunk 1 above)
Comment (iaulakh) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/JoinCallConfig.java]: Thanks Wayne. also changed function input parameters name.

**Review by waynemo — CHANGES_REQUESTED** [review]
I think we can keep the JoinCallType enum changes as well. We can just have GROUP_CALL for now.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 2]
```diff
@@ -363,8 +363,8 @@ private void openShareDialogue() {
         Log.d(LOG_TAG, "Share button clicked!");
         final Intent sendIntent = new Intent();
         sendIntent.setAction(Intent.ACTION_SEND);
-        sendIntent.putExtra(Intent.EXTRA_TEXT, callingContext.getGroupId());
-        sendIntent.putExtra(Intent.EXTRA_TITLE, "Group Call ID");
+        sendIntent.putExtra(Intent.EXTRA_TEXT, callingContext.getJoinId());
+        sendIntent.putExtra(Intent.EXTRA_TITLE, "Join Call ID");
```
[the remark this answers is not in the document]
Comment (iaulakh) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java]: ok

**Comment by iaulakh** [comment]
> he JoinCallType enum changes as well. We can just have GROUP_CALL for now.

ok

**Review by waynemo — APPROVED** [review]

**Review by lpeixin — APPROVED** [review]

**Pull request merged**